File: vod/src/main/java/top/cywin/onetv/vod/player/exo/CacheManager.java
package top.cywin.onetv.vod.player.exo;

import androidx.media3.database.StandaloneDatabaseProvider;
import androidx.media3.datasource.cache.Cache;
import androidx.media3.datasource.cache.NoOpCacheEvictor;
import androidx.media3.datasource.cache.SimpleCache;

import top.cywin.onetv.vod.App;
import top.github.catvod.utils.Path;

public class CacheManager {

    private SimpleCache cache;

    private static class Loader {
        static volatile CacheManager INSTANCE = new CacheManager();
    }

    public static CacheManager get() {
        return Loader.INSTANCE;
    }

    public Cache getCache() {
        if (cache == null) create();
        return cache;
    }

    private void create() {
        cache = new SimpleCache(Path.exo(), new NoOpCacheEvictor(), new StandaloneDatabaseProvider(App.get()));
    }
}

File: movie/src/main/java/top/cywin/onetv/movie/data/cache/CacheManager.kt
package top.cywin.onetv.movie.data.cache

/**
 * 缓存管理器接口
 * 定义缓存操作的标准接口
 */
interface CacheManager {
    /**
     * 获取缓存
     */
    suspend fun get(key: String): String?
    
    /**
     * 设置缓存
     */
    suspend fun put(key: String, value: String, vararg ttl: Long)
    
    /**
     * 清除缓存
     */
    suspend fun clear()
    
    /**
     * 获取缓存大小
     */
    suspend fun getSize(): Long
}
